lib/pure/collections/deques.nim

```diff
@@ -51,46 +51,55 @@ runnableExamples:
 
 import std/private/since
 
-import math
+import std / [math, hashes]
 
 type
   Deque*[T] = object
     ## A double-ended queue backed with a ringed `seq` buffer.
     ##
-    ## To initialize an empty deque,
-    ## use the `initDeque proc <#initDeque,int>`_.
+    ## To initialize an empty deque with a given capacity use
+    ## `initDeque proc <#initDeque,int>`_.
     data: seq[T]
     head, tail, count, mask: int
 
 const
-  defaultInitialSize* = 4
+  nimDequeDefaultInitialCapacity* {.intdefine.} = 4 # Maybe this should be nimDefaultInitialCapacity as well?
 
-template initImpl(result: typed, initialSize: int) =
-  let correctSize = nextPowerOfTwo(initialSize)
-  result.mask = correctSize - 1
-  newSeq(result.data, correctSize)
+template initImpl(result: typed, initialCapacity: int) =
+  assert isPowerOfTwo(initialCapacity)
+  result.mask = initialCapacity - 1
+  newSeq(result.data, initialCapacity)
 
 template checkIfInitialized(deq: typed) =
-  when compiles(defaultInitialSize):
+  when declared(nimDequeDefaultInitialCapacity):
```

Comment:
Pre-exists. But why this line is needed? It seems unnecessary. See also https://github.com/nim-works/nimskull/pull/161#issue-1105184999